### PR TITLE
Add validation to NameEvent: name should not exceed 255 bytes

### PR DESCRIPTION
**Pull request by lu-fennell**

### CODE REVIEW

**Comment by jplatte** [comment]
The CI failure is fixed now on master, it has nothing to do with your changes. However, there are a few things to improve still.

File: src/room/name.rs [hunk 1]
```diff
@@ -227,14 +223,15 @@ pub(crate) mod raw {
 #[cfg(test)]
 mod tests {
     use std::convert::TryFrom;
+    use std::iter::FromIterator;
```
Comment (jplatte) [inline src/room/name.rs]: You can nest these imports: `use std::{convert::TryFrom, iter::FromIterator};`

Comment (lu-fennell) [reply src/room/name.rs]: Done

File: src/room/name.rs [hunk 2]
```diff
@@ -251,6 +265,42 @@ mod tests {
...
     }
 
+    #[test]
+    fn name_fails_validation_when_too_long() {
+        // "XXXX .." 256 times
+        let long_string: String = String::from_iter(std::iter::repeat('X').take(256));
+        assert_eq!(long_string.len(), 256);
+
+        let long_content_json_string: String =
+            serde_json::json!({ "name": long_string.as_str()}).to_string();
```
Comment (jplatte) [inline src/room/name.rs]:
```suggestion
            serde_json::json!({ "name": long_string.as_str() }).to_string();
```

File: src/room/name.rs [hunk 3]
```diff
@@ -251,6 +265,42 @@ mod tests {
...
+        let from_raw: EventResult<NameEventContent> =
+            serde_json::from_str(long_content_json_string.as_str()).unwrap();
+
+        let result = from_raw.into_result();
+        assert!(result.is_err(), "Result should be invalid: {:?}", result);
+    }
+
+    #[test]
+    fn json_with_empty_name_creates_content_as_none() {
+        let long_content_json_string: String = serde_json::json!({ "name": ""}).to_string();
```
Comment (jplatte) [inline src/room/name.rs]:
```suggestion
        let long_content_json_string: String = serde_json::json!({ "name": "" }).to_string();
```

File: src/room/name.rs [hunk 4]
```diff
@@ -43,28 +43,41 @@ pub struct NameEventContent {
...
     type Raw = raw::NameEvent;
+    type Err = InvalidInput;
 
-    fn from_raw(raw: raw::NameEvent) -> Self {
-        Self {
-            content: FromRaw::from_raw(raw.content),
+    fn try_from_raw(raw: Self::Raw) -> Result<Self, Self::Err> {
+        let content = TryFromRaw::try_from_raw(raw.content)?;
+
+        let prev_content = match raw.prev_content {
```
Comment (jplatte) [inline src/room/name.rs]: You can use [Result::transpose](https://doc.rust-lang.org/std/result/enum.Result.html#method.transpose) here :)

File: src/room/name.rs (hunk 2 above)
Comment (lu-fennell) [reply src/room/name.rs]: Done. And now also `&long_string`.

File: src/room/name.rs (hunk 3 above)
Comment (lu-fennell) [reply src/room/name.rs]: Done

File: src/room/name.rs (hunk 4 above)
Comment (lu-fennell) [reply src/room/name.rs]: Cool!

**Pull request merged**